src/main/java/net/dries007/tfc/common/blocks/crop/ICropBlock.java

```diff
@@ -34,4 +37,10 @@ default float getGrowthLimit(Level level, BlockPos pos, BlockState state)
     ClimateRange getClimateRange();
 
     FarmlandBlockEntity.NutrientType getPrimaryNutrient();
+
+    float getGrowthFactor(Level level, BlockPos pos, BlockState state, CropBlockEntity crop);
+
+    float getNutrientConsumptionFactor(Level level, BlockPos pos, BlockState state, CropBlockEntity crop);
+
+    float getNutrientResupplyFactor(Level level, BlockPos pos, BlockState state, CropBlockEntity crop);
```

Comment:
Even more breaking changes. These methods do not need to exist, they do not need the level of context given to them, it's ridiculous that they would be an API. They're completely undocumented and just... why.